src/bundle/Controller/Content/ContentTreeController.php

```diff
@@ -78,13 +79,14 @@ public function loadChildrenAction(
         int $parentLocationId,
         int $limit,
         int $offset,
-        Query\Criterion $filter
+        FilterQueryCriteria $filter
     ): Node {
         $location = $this->locationService->loadLocation($parentLocationId);
         $loadSubtreeRequestNode = new LoadSubtreeRequestNode($parentLocationId, $limit, $offset);
 
         $sortClause = $request->query->get('sortClause', null);
         $sortOrder = $request->query->getAlpha('sortOrder', Query::SORT_ASC);
+        $filter = $filter->getCriteria();
```

Comment:
This class looks suspiciously similar to `LogicalAnd`, to be honest.

Maybe we should just typehint against `LogicalAnd` instead? And just check how many criteria are inside? That would make it unnecessary to have yet another class for a criteria collection.

Reply:
I tried to avoid checking criteria inside LogicalAnd, but if it is acceptable solution I can change it.

Reply:
https://github.com/ibexa/admin-ui/pull/1328/commits/e5d17891e2eafb7e335808651910434989e1a9c7